qcengine/programs/qchem.py

```diff
@@ -0,0 +1,251 @@
+            envs["QCSCRATCH"] = tmpdir
+            bdict = {x: None for x in binary_files}
+
+            with disk_files({}, bdict, cwd=tmpdir, as_binary=binary_files):
+                exe_success, proc = execute(commands,
+                                            infiles=infiles,
+                                            outfiles=outfiles,
+                                            scratch_name=scratch_name,
+                                            scratch_directory=tmpdir,
+                                            scratch_messy=scratch_messy,
+                                            timeout=timeout,
+                                            environment=envs)
+
+            proc["outfiles"].update({os.path.split(k)[-1]: v for k, v in bdict.items()})
+
+        if "Thank you very much for using Q-Chem" not in proc["outfiles"]["dispatch.out"]:
+            exe_success = False
+
+        # QChem does not create an output file and only prints to stdout
+        return exe_success, proc
+
+    def build_input(self, input_model: 'ResultInput', config: 'JobConfig',
+                    template: Optional[str] = None) -> Dict[str, Any]:
+
+        # Check some bounds on what cannot be parsed
+        if "ccsd" in input_model.method.lower() or "ccd" in input_model.method.lower():
+            raise InputError("Cannot handle CC* methods currently.")
+
+        # Build keywords
+        keywords = {k.upper(): v for k, v in input_model.keywords.items()}
+        keywords["INPUT_BOHR"] = "TRUE"
+        keywords["MEM_TOTAL"] = str(int(config.memory * 1024))  # In MB
+
+        if input_model.driver == "energy":
+            keywords["JOBTYPE"] = "sp"
+        elif input_model.driver == "gradient":
+            keywords["JOBTYPE"] = "force"
+        elif input_model.driver == "hessian":
+            keywords["JOBTYPE"] = "freq"
+        else:
+            raise InputError(f"Driver of type {input_model.driver} is not yet supported.")
+
+        if input_model.molecule.fix_com or input_model.molecule.fix_orientation:
+            keywords["NO_REORIENT"] = "TRUE"
+
+        keywords["METHOD"] = input_model.model.method
+        if input_model.model.basis:
+            keywords["BASIS"] = input_model.model.basis
+
+        # Begin the input file
+        input_file = []
+        input_file.append(f"""$comment
+Automatically generated Q-Chem input file by QCEngine
+$end
+            """)
+
+        # Add Molecule, TODO: Add to QCElemental
+        mol = input_model.molecule
+        input_file.append("$molecule")
+        input_file.append(f"""{int(mol.molecular_charge)} {mol.molecular_multiplicity}""")
+
+        for real, sym, geom in zip(mol.real, mol.symbols, mol.geometry):
+            if real is False:
+                raise InputError("Cannot handle ghost atoms yet.")
+            input_file.append(f"{sym} {geom[0]:14.8f} {geom[1]:14.8f} {geom[2]:14.8f}")
+
+        input_file.append("$end\n")
```

Comment:
once MolSSI/QCElemental#157 goes in, should be able to remove this in favor of https://github.com/MolSSI/QCEngine/blob/master/qcengine/programs/cfour/runner.py#L92-L93